src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -115,11 +115,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $userClass = $this->getUserClass($input, $io);
         $emptySalt = $input->getOption('empty-salt');
 
-        $encoderFactory = $this->encoderFactory ?: $this->getContainer()->get('security.encoder_factory');
```

Comment:
This is a BC layer, it must be kept

Reply:
Can you revert to keep the `else`? This is what breaks tests

Reply:
Sorry about that. Changed it back.